tensorflow/python/ops/bitwise_ops_test.py

```diff
@@ -135,5 +135,31 @@ def testImplementationDefinedShiftsDoNotCrash(self):
                   bitwise_ops.right_shift(lhs, rhs)])
 
 
+  def testShapeInference(self):
+    dtype_list = [dtypes.int8, dtypes.int16, dtypes.int32, dtypes.int64,
+                  dtypes.uint8, dtypes.uint16]
+
+    with self.test_session(use_gpu=True) as sess:
+      for dtype in dtype_list:
+        lhs = constant_op.constant([[0], [3], [5]], dtype=dtype)
+        rhs = constant_op.constant([[1, 2, 4]], dtype=dtype)
+
+        and_tensor = bitwise_ops.bitwise_and(lhs, rhs)
+        or_tensor = bitwise_ops.bitwise_or(lhs, rhs)
+        xor_tensor = bitwise_ops.bitwise_xor(lhs, rhs)
+        ls_tensor = bitwise_ops.left_shift(lhs, rhs)
+        rs_tensor = bitwise_ops.right_shift(lhs, rhs)
+
+        and_result, or_result, xor_result, ls_result, rs_result = sess.run(
+            [and_tensor, or_tensor, xor_tensor, ls_tensor, rs_tensor])
+
+        # Compare shape inference with result
+        self.assertAllEqual(and_tensor.get_shape().as_list(), and_result.shape)
```

Comment:
I think `[3, 3]` might be better than `and_result.shape`

Reply:
@facaiy `and_result.shape` is to make sure the shape inference beforethe  `session.run` is the same as the final shape after evaluated. This is to maintain consistency before and after eval.

Reply:
I shouldn't nitpick, however `and_result.shape` might not be the result we expected, say, there exits a bug in the route.  Anyway, I think both are accepted.